connect-inject/envoy_sidecar_test.go

```diff
@@ -203,3 +204,187 @@ func TestHandlerEnvoySidecar_NamespacesAndAuthMethod(t *testing.T) {
 && /consul/connect-inject/consul logout \
   -token-file="/consul/connect-inject/acl-token"`)
 }
+
+func TestHandlerEnvoySidecar_Resources(t *testing.T) {
+	mem1, err := resource.ParseQuantity("100Mi")
+	require.NoError(t, err)
+	mem2, err := resource.ParseQuantity("200Mi")
+	require.NoError(t, err)
+	cpu1, err := resource.ParseQuantity("100m")
+	require.NoError(t, err)
+	cpu2, err := resource.ParseQuantity("200m")
+	require.NoError(t, err)
+	zero, err := resource.ParseQuantity("0")
+	require.NoError(t, err)
+
+	cases := map[string]struct {
+		handler      Handler
+		annotations  map[string]string
+		expResources corev1.ResourceRequirements
+		expErr       string
+	}{
+		"no defaults, no annotations": {
+			handler:     Handler{},
+			annotations: nil,
+			expResources: corev1.ResourceRequirements{
+				Limits:   corev1.ResourceList{},
+				Requests: corev1.ResourceList{},
+			},
+		},
+		"all defaults, no annotations": {
+			handler: Handler{
+				DefaultProxyCPURequest:    cpu1,
+				DefaultProxyCPULimit:      cpu2,
+				DefaultProxyMemoryRequest: mem1,
+				DefaultProxyMemoryLimit:   mem2,
+			},
+			annotations: nil,
+			expResources: corev1.ResourceRequirements{
+				Limits: corev1.ResourceList{
+					corev1.ResourceCPU:    cpu2,
+					corev1.ResourceMemory: mem2,
+				},
+				Requests: corev1.ResourceList{
+					corev1.ResourceCPU:    cpu1,
+					corev1.ResourceMemory: mem1,
+				},
+			},
+		},
+		"no defaults, all annotations": {
+			handler: Handler{},
+			annotations: map[string]string{
+				annotationSidecarProxyCPURequest:    "100m",
+				annotationSidecarProxyMemoryRequest: "100Mi",
+				annotationSidecarProxyCPULimit:      "200m",
+				annotationSidecarProxyMemoryLimit:   "200Mi",
+			},
+			expResources: corev1.ResourceRequirements{
+				Limits: corev1.ResourceList{
+					corev1.ResourceCPU:    cpu2,
+					corev1.ResourceMemory: mem2,
+				},
+				Requests: corev1.ResourceList{
+					corev1.ResourceCPU:    cpu1,
+					corev1.ResourceMemory: mem1,
+				},
+			},
+		},
+		"annotations override defaults": {
+			handler: Handler{
+				DefaultProxyCPURequest:    zero,
+				DefaultProxyCPULimit:      zero,
+				DefaultProxyMemoryRequest: zero,
+				DefaultProxyMemoryLimit:   zero,
+			},
+			annotations: map[string]string{
+				annotationSidecarProxyCPURequest:    "100m",
+				annotationSidecarProxyMemoryRequest: "100Mi",
+				annotationSidecarProxyCPULimit:      "200m",
+				annotationSidecarProxyMemoryLimit:   "200Mi",
+			},
+			expResources: corev1.ResourceRequirements{
+				Limits: corev1.ResourceList{
+					corev1.ResourceCPU:    cpu2,
+					corev1.ResourceMemory: mem2,
+				},
+				Requests: corev1.ResourceList{
+					corev1.ResourceCPU:    cpu1,
+					corev1.ResourceMemory: mem1,
+				},
+			},
+		},
+		"defaults set to zero, no annotations": {
```

Comment:
Do you think this is a valid value we should accept?

Reply:
Yes it's a valid valid that means no limit. I think we should accept it because someone may want to set it explicitly to 0 in an environment that requires it be set (think maybe a admission webhook).